File: C#/toys_shop/Dal/models/Customer.cs
﻿using System;
using System.Collections.Generic;

namespace Dal.models;

public partial class Customer
{
    public int CustomerId { get; set; }

    public string CustomerName { get; set; } = null!;

    public string CustomerPhone { get; set; } = null!;

    public string CustomerEmail { get; set; } = null!;

    public string CustomerPassword { get; set; } = null!;

    public DateOnly CustomerDateOfBirth { get; set; }

    public virtual ICollection<Purchase> Purchases { get; set; } = new List<Purchase>();
}

File: ANGULAR/src/app/models/customer.ts
export class Customer {
    customerId: number;
    customerName: string;
    customerPhoneNumber: string;
    customerEmail: string;
    customerPassword:string;
    customerDateOfBirth: Date;

    constructor(customerId: number, customerName: string, customerPhoneNumber: string, customerEmail: string, customerDateOfBirth: Date,customerPassword:string) {
        this.customerDateOfBirth = customerDateOfBirth;
        this.customerEmail = customerEmail;
        this.customerId = customerId;
        this.customerName = customerName;
        this.customerPhoneNumber = customerPhoneNumber;
        this.customerPassword=customerPassword;
    }
}
